tests/integration2/galley/conversion/Readme.md

```diff
@@ -0,0 +1,65 @@
+**How To Use Galley Conversion Tests**
+
+The Galley Conversion Tests are designed to tests Galley from an inputs/outputs
+perspective. It uses an embedded set of input and golden files from which
+tests are calculated and executed.
+
+The directory structure is as follows:
+
+```
+# Top Level Folder for the Test
+conversion/
+
+# The main driver code for the test
+conversion/conversion_test.go
+conversion/main_test.go
+
+# Go Library that contains the test data.
+conversion/testdata
+
+# The actual dataset folder
+conversion/testdata/dataset
+```
+
+Typically, a set of files needed for a test is as follows:
+
+```
+# Input file for the test
+.../dataset/**/<testname>.yaml
+
+# Optional Mesh Config Input file
+.../dataset/**/<testname>_meshconfig.yaml
+
+# Required expected resources file.
+.../dataset/**/<testname>_expected.json
```

Comment:
What is the reason mixing YAML for input and JSON for output?

Reply:
Yaml is the typical input file format we have that works against Kubernetes. Using any other format, such as JSON will be awkward.

JSON is one of the canonical/official textual representations of a Proto. Using Yaml will require us to do a translation back to JSON for eventual conversion to Proto. There is potential to introduce bugs. Using JSON makes sure that we can do strict input/output testing without having to do unneccessary transformations.